### PR TITLE
Add node to scale units using enginering prefixes

### PR DESCRIPTION
**Pull request by jenshnielsen**
Based on the logic implemented in qcodes. The logic in qcodes has been refactored in https://github.com/QCoDeS/Qcodes/pull/2480
to support older versions we bundle the same code here for now until older versions of qcodes are no longer supported.

### CODE REVIEW

File: plottr/node/scaleunits.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    #: do not scale units
+    never = 0
+
+    #: all units
+    always = 1
+
+
+class ScaleUnitOptionWidget(QtWidgets.QWidget):
+
+    unitscaleselected = Signal(ScaleUnitsOption)
```
Comment (astafan8) [inline plottr/node/scaleunits.py]:
```suggestion
    unit_scale_selected = Signal(ScaleUnitsOption)
```

**Review by astafan8 — COMMENTED** [review]
Appreciate this as a "add a node" excercise, saves me some time realizing what nodes and widgets are needed.

File: plottr/node/scaleunits.py (hunk 1 above)
Comment (jenshnielsen) [reply plottr/node/scaleunits.py]: Done in a commit to rename all uses

**Comment by jenshnielsen** [comment]
To be consistent with qcodes https://github.com/QCoDeS/Qcodes/pull/2488 should be merged too

File: plottr/node/scaleunits.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        self._scale_unit_option = value
+
+    def process(self, dataIn: Optional[DataDictBase] = None) -> Optional[Dict[str, Optional[DataDictBase]]]:
+        if super().process(dataIn=dataIn) is None:
+            return None
+        assert dataIn is not None
+        data = dataIn.copy()
+
+        for name, data_item in data.data_items():
+            if self.scale_unit_option != ScaleUnitsOption.never:
```
Comment (astafan8) [inline plottr/node/scaleunits.py]: perhaps a more readable order of loops?
```suggestion
        if self.scale_unit_option != ScaleUnitsOption.never:
            for name, data_item in data.data_items():        
```

File: plottr/node/scaleunits.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                prefix, selected_scale = find_scale_and_prefix(
+                    data_item['values'],
+                    data_item["unit"]
+                )
+                data_item["unit"] = prefix + data_item["unit"]
+                data_item['values'] = data_item['values'] * 10**(-selected_scale)
```
Comment (astafan8) [inline plottr/node/scaleunits.py:134]: i remember there was a comment about categorical data - is it out of scope or did it get resloved? if i remember correctly, in qcodes, categorical data does not get unit-scaled, so perhaps we should do the same here?

**Review by astafan8 — APPROVED** [review]

File: plottr/node/scaleunits.py (hunk 3 above)
Comment (jenshnielsen) [reply plottr/node/scaleunits.py:134]: Plottr does not in general support categorical data

**Pull request merged**